packages/fiori-generator-shared/src/telemetry/utils.ts

```diff
@@ -0,0 +1,67 @@
+import { ClientFactory, PerformanceMeasurementAPI as Performance, SampleRate } from '@sap-ux/telemetry';
+import type { TelemetryEvent } from '@sap-ux/telemetry';
+import type { TelemetryData } from './types';
+import { TelemetryHelper } from './telemetryHelper';
+
+/**
+ * Prepares the telemetry event to be sent to the telemetry client.
+ *
+ * @param generationEvent - the event name to be reported
+ * @param telemetryData - the telemetry data
+ * @returns - the telemetry event
+ */
+function prepareTelemetryEvent(generationEvent: string, telemetryData: TelemetryData): TelemetryEvent {
+    // Make sure performance measurement end is called
+    TelemetryHelper.markAppGenEndTime();
+    const generationTime = telemetryData.markName
+        ? Performance.getMeasurementDuration(telemetryData.markName)
+        : undefined;
+
+    return {
+        eventName: generationEvent,
+        properties: telemetryData,
+        measurements: generationTime ? { GenerationTime: generationTime } : {}
+    };
+}
+
+/**
+ * Sends the telemetry event to the telemetry client.
+ *
+ * @param generationEvent - the event name to be reported
```

Comment:
This could be any event, as its defined as a stirng?

Reply:
Fair point, I've updated to be `telemetryEventName`